js/search.js

```diff
@@ -962,11 +961,39 @@
 				if (id === 'hiddenpowergrass') isViable = (moves.indexOf('energyball') < 0 && moves.indexOf('gigadrain') < 0);
 				if (id === 'hiddenpowerice') isViable = (moves.indexOf('icebeam') < 0 && template.id !== 'xerneas');
 				if (id === 'icywind') isViable = (toId(set.species).substr(0, 6) === 'keldeo');
+				if (id === 'leechlife') isViable = (this.gen > 6);
 				if (id === 'reflecttype') isViable = (toId(set.species) in {latias:1, starmie:1});
 				if (id === 'selfdestruct') isViable = (this.gen < 5 && moves.indexOf('explosion') < 0);
 				if (id === 'skyattack') isViable = (toId(set.species) === 'hawlucha');
 				if (id === 'stunspore') isViable = (moves.indexOf('thunderwave') < 0);
-				if (id === 'toxic') isViable = (this.gen > 1);
+				if (this.gen === 1) {
+					// Usually viable for Gen 1
+					if (id === 'acidarmor' || id === 'amnesia' || id === 'barrier' || id === 'bind' || id === 'clamp' || id === 'confuseray' || id === 'counter' || id === 'firespin' || id === 'hyperbeam' || id === 'mirrormove' || id === 'pinmissile' || id === 'razorleaf' || id === 'sing' || id === 'slash' || id === 'sludge' || id === 'twineedle' || id === 'wrap') isViable = true;
+
+					// Usually not viable for Gen 1
+					if (id === 'disable' || id === 'firepunch' || id === 'icepunch' || id === 'jumpkick' || id === 'leechseed' || id === 'quickattack' || id === 'roar' || id === 'thunderpunch' || id === 'toxic' || id === 'triattack' || id === 'whirlwind') isViable = false;
+
+					// Viable only when certain moves aren't present
+					if (id === 'doubleedge') isViable = (moves.indexOf('bodyslam') < 0);
+					if (id === 'hypnosis') isViable = (moves.indexOf('sleeppowder') < 0);
+					if (id === 'megadrain') isViable = (moves.indexOf('razorleaf') < 0 && moves.indexOf('surf') < 0);
+					if (id === 'megakick') isViable = (moves.indexOf('hyperbeam') < 0);
+					if (id === 'reflect') isViable = (moves.indexOf('barrier') < 0 && moves.indexOf('acidarmor') < 0);
+
+					// Viable for specific types
+					if (id === 'substitute') isViable = (template.types[0] === 'Ground' || template.types[1] === 'Ground');
```

Comment:
Works like later gen Substitutes, except doesn't block status (except in Gen 1 Stadium, which has differing mechanics to normal RBY). Ground-types have a much better time using Substitute because:
1) All Ground-type generally have all the coverage and power they need in Rock Slide + Earthquake leaving them with an easy time fitting Substitute on a set.
2) Pokemon in RBY are usually crippled by status, and most Pokemon that inflict status do so with Thunder Wave which Ground-types are immune to. This reduces the problem with status moves going through Substitute.
3) Ground-types are often very weak to common Water/Ice/Grass attacks that can 1-2HKO, and so successfully finding an opening to use Substitute (such as fishing for Full Paralysis, scaring out an opponent, setting up on a sleeping/frozen Pokemon) can give them an extra turn they wouldn't have had otherwise to hammer the opponent with powerful STAB Earthquakes.